app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.java

```diff
@@ -41,14 +40,13 @@
 
 public class WebServerHelper {
   private static final String TAG = "WebServerHelper";
-  private Library kiwixLibrary;
-  private JNIKiwixServer kiwixServer;
+  private KiwixServer kiwixServer;
```

Comment:
The idea behind KiwixServer was that it would be a facade for JNIKiwixServer, right now you are using KiwixServer as a factory for JniKiwixServer and I'd rather there was a KiwixServer.Factory that creates `KiwixServer`s which have methods `start(port)` & `stop()`

Reply:
If I create a start & stop function in `KiwixServer` class, then it'll have to take `server` as a param too or we can avoid that via creating an extension fun on server

Reply:
KiwixServer will be a class that accepts JNIKiwixServer as a constructor val

Reply:
How's that going to help me, when I'll create a new start function in `KiwixServer`, I won't be able to access the constructor param, then how?

Reply:
Is there any problem/violation with the usage of extension function?

Reply:
you can access the constructor param, as I said in another comment on this PR eg

```
class KiwixServer(private val jniKiwixServer:JniKiwiServer){

fun stop(){
  jniKiwixServer.stop()
}
```

Reply:
The point of KiwixServer is to be a one stop class that explains this coupled nature of Library/JniKiwixServer if somebody cares to look but from the outside has a simple API that a reader can just breeze through.

Technically I guess KiwixServer.Factory could really just be JniKiwixServerFactory as the Libray with books is just an implementation detail but I prefer having these wrapper classes around our JNI objects, sometimes kiwixlib updates internal representations and it is useful to only have to update in 1 place, not that it matters tooooo much on this one as we only use server in this 1 place, still good practice though.

Reply:
Ah shoot! missed it

Reply:
>  KiwixServer will be a class that accepts JNIKiwixServer as a constructor val

If we consider it logically, then we're using `Factory` class inside `KiwixServer` class to create `JNIKiwixServer` and the main class itself needs a `JNIKiwixServer` as a param to create an instance in some other java class, whoah! Sounds extremely weird to me.